src/main/java/io/stargate/sgv2/jsonapi/service/shredding/model/WritableShreddedDocument.java

```diff
@@ -39,7 +43,53 @@ public record WritableShreddedDocument(
     Map<JsonPath, BigDecimal> queryNumberValues,
     Map<JsonPath, String> queryTextValues,
     Map<JsonPath, Date> queryTimestampValues,
-    Set<JsonPath> queryNullValues) {
+    Set<JsonPath> queryNullValues,
+    float[] queryVectorValues) {
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
Are equality and hashCode() needed for something, or just auto-generated?

Reply:
They are auto generated, without this the equals on float[] field doesn't match

Reply:
Ok so why were these needed (did not exist before)? Just curious, nothing wrong with impl or so.